src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -839,20 +838,32 @@ public unsafe ContentId GetContentId()
         {
             if (_deterministic)
             {
-                // Call to GetDebugInfo fails for SymWriter initialized using InitializeDeterministic.
-                // We already have all the info we need though.
-                var id = _callLogger.ContentIdFromLog();
+                // rewrite GUID and timestamp in the PDB with hash of a has of the log content:
+                byte[] hash = _callLogger.GetLogHash();
+
                 try
                 {
-                    Debug.Assert(BitConverter.IsLittleEndian);
-                    ((ISymUnmanagedWriter100)_symWriter).SetSignature(BitConverter.ToUInt32(id.Stamp, 0), new Guid(id.Guid));
+                    // TODO: remove once we can rely on the presence of ISymUnmanagedWriter7
```

Comment:
ISymUnmanagedWriter100 instead of ISymUnmanagedWriter7? The new one is the one with 100, right?